File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/view/dialogs/NewGroupIncompleteErrorDialog.java
package it.dani.selfhomeapp.middle.view.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.function.Supplier;

import it.dani.selfhomeapp.R;

public class NewGroupIncompleteErrorDialog extends DialogFragment {

    private Supplier<Void> negativeSupplier;

    public NewGroupIncompleteErrorDialog()
    {
        super();
        this.negativeSupplier = null;
    }

    public NewGroupIncompleteErrorDialog(@NonNull Supplier<Void> negativeSupplier)
    {
        super();
        this.negativeSupplier = negativeSupplier;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstance)
    {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        builder.setTitle(R.string.titleNewGroupIncompleteResponse);
        builder.setMessage(R.string.newGroupIncompleteResponse);

        builder.setNegativeButton(R.string.negativeNewGroupIncompleteResponse,(x,y) -> {
            if(this.negativeSupplier != null) this.negativeSupplier.get();
        });

        return builder.create();
    }
}

File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/view/dialogs/NewGroupIncompleteErrorDialog.kt
package it.dani.selfhomeapp.middle.view.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import it.dani.selfhomeapp.R

class NewGroupIncompleteErrorDialog(private val negativeSupplier : () -> Unit) : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(this.activity).apply {
            setTitle(R.string.titleNewGroupIncompleteResponse)
            setMessage(R.string.newGroupIncompleteResponse)
            setNegativeButton(R.string.negativeNewGroupIncompleteResponse) { _, _ ->
                this@NewGroupIncompleteErrorDialog.negativeSupplier()
            }
        }.create()
    }
}